.github/workflows/system-tests.yml

```diff
@@ -10,28 +10,40 @@ on:
 
 env:
   REGISTRY: ghcr.io
+  REPO: ghcr.io/datadog/dd-trace-rb
 
 jobs:
   build-harness:
     strategy:
       fail-fast: false
       matrix:
         image:
-          - agent
+          - name: runner
+            internal: system_tests/runner:latest
+          - name: agent
+            internal: system_tests/agent:latest
+          - name: proxy
+            internal: datadog/system-tests:proxy-v1
     runs-on: ubuntu-latest
-    name: Build (${{ matrix.image }})
+    name: Build (${{ matrix.image.name }})
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           repository: 'DataDog/system-tests'
       - name: Pull released image
         run: |
-          if docker pull ghcr.io/datadog/dd-trace-rb/system-tests/${{ matrix.image }}:latest; then
-            docker tag ghcr.io/datadog/dd-trace-rb/system-tests/${{ matrix.image }}:latest system_tests/${{ matrix.image }}:latest
+          if docker pull ${{ env.REPO }}/system-tests/${{ matrix.image.name }}:latest; then
+            docker tag ${{ env.REPO }}/system-tests/${{ matrix.image.name }}:latest ${{ matrix.image.internal }}
           fi
```

Comment:
Eve though `--cache-from` has been added this is still needed because of the special internal naming of system tests, for which a change is planned to be proposed. See https://github.com/DataDog/dd-trace-rb/pull/3543#discussion_r1535377150